packages/wpcom-checkout/src/partner-coupon.ts

```diff
@@ -0,0 +1,16 @@
+import { getSublabel } from './checkout-labels';
+import type { ResponseCartProduct } from '@automattic/shopping-cart';
+
+export function getPartnerCoupon( {
+	coupon,
+	products,
+}: {
+	coupon: string;
+	products?: ResponseCartProduct[];
+} ): boolean {
+	const productHasSublabel =
```

Comment:
_Just an observation but IMO it's currently implemented the expected way:_

Technically we have a list of supported partner products in `client/jetpack-connect/constants.js::JETPACK_COUPON_PRESET_MAPPING` - but I don't think it's necessary to check against that list because even if the cart contains an invalid partner product, we still want to show that the coupon belongs to a partner, so they can try and reach out to their support about what's going on.

TL;DR: Looks good 👍